lib/internal/util/parse_args/parse_args.js

```diff
@@ -0,0 +1,268 @@
+      StringPrototypeIncludes(execArgv, '-p') ||
+      StringPrototypeIncludes(execArgv, '--print')) {
+    return ArrayPrototypeSlice(process.argv, 1);
+  }
+
+  // Normally first two arguments are executable and script, then CLI arguments
+  return ArrayPrototypeSlice(process.argv, 2);
+}
+
+const protoKey = '__proto__';
+
+function storeOption({
+  strict,
+  options,
+  result,
+  longOption,
+  shortOption,
+  optionValue,
+}) {
+  const hasOptionConfig = ObjectHasOwn(options, longOption);
+  const optionConfig = hasOptionConfig ? options[longOption] : {};
+
+  if (strict) {
+    if (!hasOptionConfig) {
+      throw new ERR_PARSE_ARGS_UNKNOWN_OPTION(shortOption == null ? `--${longOption}` : `-${shortOption}`);
+    }
+
+    const shortOptionErr = ObjectHasOwn(optionConfig, 'short') ? `-${optionConfig.short}, ` : '';
+
+    if (options[longOption].type === 'string' && optionValue == null) {
+      throw new ERR_PARSE_ARGS_INVALID_OPTION_VALUE(`Option '${shortOptionErr}--${longOption} <value>' argument missing`);
+    }
+
+    if (options[longOption].type === 'boolean' && optionValue != null) {
+      throw new ERR_PARSE_ARGS_INVALID_OPTION_VALUE(`Option '${shortOptionErr}--${longOption}' does not take an argument`);
+    }
+  }
+
+  if (longOption === protoKey) {
+    return;
+  }
+
+  // Values
+  const usedAsFlag = optionValue === undefined;
+  const newValue = usedAsFlag ? true : optionValue;
+  if (optionConfig.multiple) {
+    // Always store value in array, including for flags.
+    // result.values[longOption] starts out not present,
+    // first value is added as new array [newValue],
+    // subsequent values are pushed to existing array.
+    if (result.values[longOption] !== undefined)
+      ArrayPrototypePush(result.values[longOption], newValue);
+    else
+      result.values[longOption] = [newValue];
+  } else {
+    result.values[longOption] = newValue;
+  }
+}
+
+const parseArgs = ({
+  args = getMainArgs(),
+  strict = true,
+  options = {}
+} = {}) => {
+  validateArray(args, 'args');
+  validateBoolean(strict, 'strict');
+  validateObject(options, 'options');
+  ArrayPrototypeForEach(
+    ObjectEntries(options),
+    ({ 0: longOption, 1: optionConfig }) => {
+      validateObject(optionConfig, `options.${longOption}`);
+
+      validateUnion(optionConfig.type, `options.${longOption}.type`, ['string', 'boolean']);
+
+      if (ObjectHasOwn(optionConfig, 'short')) {
+        const shortOption = optionConfig.short;
+        validateString(shortOption, `options.${longOption}.short`);
+        if (shortOption.length !== 1) {
+          throw new ERR_INVALID_ARG_VALUE(
+            `options.${longOption}.short`,
+            shortOption,
+            'must be a single character'
+          );
+        }
+      }
+
+      if (ObjectHasOwn(optionConfig, 'multiple')) {
+        validateBoolean(optionConfig.multiple, `options.${longOption}.multiple`);
+      }
+    }
+  );
+
+  const result = {
+    values: {},
```

Comment:
This one isn't just paranoia about prototype pollution: without it, code like

```js
let { values } = parseArgs({ toString: { type: 'boolean' }, args: [] });

if (values.toString) {
  // etc
}
```
is outright broken.

Reply:
Using `__proto__` relies on the globally mutable `set %Object.prototype%.__proto__`:
```suggestion
    values: ObjectCreate(null),
```

Reply:
> Using `__proto__` relies on the globally mutable `set %Object.prototype%.__proto__`:

No, it doesn't. It's a special syntactic form, and does not rely on the accessor on Object.prototype. See [PropertyDefinitionEvaluation](https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-runtime-semantics-propertydefinitionevaluation) for `PropertyDefinition : PropertyName : AssignmentExpression`, steps 4 & 8.

Reply:
@ljharb @bakkot @aduh95, bother you to also make the recommendation here:

https://github.com/pkgjs/parseargs/pull/106

Then we can keep the work in sync with @shadowspawn's PR -- we should add a test for @bakkot's example.

Reply:
Done.

Reply:
We already use the `__proto__: null` in several places in core.

Reply:
A null-inherited object was discussed in https://github.com/pkgjs/parseargs/issues/32#issue-1097132251. The `values` object is returned to the user, and at the time we were wondering about copying it back to a "normal" object before returning it. 

The concern with returning a null-inherited object to the user is that they seem pretty rare in the wild, and likely to cause some confusion. For example, there is a decent block of documentation covering [Custom and Null objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create#custom_and_null_objects) and issues.

> We already use the __proto__: null syntax in several places in core.

Are any returned to the user, or are the objects only used internally?

Reply:
I don't think they're _that_ rare - every `import * as foo` is one, as is every `.groups` on a RegExp match object. In practice nobody calls Object.prototype methods on objects anyways, so they'd work identically.

Reply:
Any dictionary-like object has to have a null prototype to work correctly, as in my example above. Dictionary-like objects don't come up all that much, but as @ljharb says, null prototype objects are what the language itself uses in this case, and it's good practice in userland also.

Reply:
I understand the theory, it is whether user-land is ready for the practice. 😄  Last time I looked seriously at this was a couple of years ago though, so my "pretty rare in the wild" may be a bit dated.

Reply:
I'd be interested to see any userland code - historical or modern - for which this would impact it in a negative way.

Reply:
`console.log('args = ' + args)` throws, instead of printing `[object Object]`, is the main negative effect. (I think that's an acceptable cost, personally.)

Reply:
I consider that a positive impact :-) because nobody wants `[object Object]` logged there :-p

Reply:
Example confusion, highest rated answer says it is a feature but the others offer "fixes": https://stackoverflow.com/questions/56298481/how-to-fix-object-null-prototype-title-product

Reply:
For comparison, Commander uses plain `{}` for storing the option values. However, that was a big step forward from the original implementation which was properties directly added to a Command extending from EventEmitter, which did cause some name conflicts people noticed! I also considered Map and null prototype at the time: https://github.com/tj/commander.js/pull/1102#issue-527469955)

Reply:
https://github.com/pkgjs/parseargs/issues/108